src/editor.rs

```diff
@@ -729,7 +724,7 @@ impl Editor {
             self.file_name = None;
         }
         loop {
-            if let Some(mode) = self.prompt_mode.as_ref() {
+            if let Some(ref mode) = self.prompt_mode.as_ref() {
```

Comment:
Is this needed? See clippy error in the PR checks

Reply:
oops - probably not - one or the other is needed. I think I was experimenting with it and didn't commit version with just `ref mode`

Reply:
fe62086...f88e37f